content/chapters/compute/lab/support/user-level-threads/threads_and_fibers.cc

```diff
@@ -0,0 +1,90 @@
+#include <vector>
+#include <thread>
+
+#include <boost/fiber/all.hpp>
+#include <boost/fiber/detail/thread_barrier.hpp>
+
+#define NUM_FIBERS 10
+#define NUM_THREADS 3
+
+static boost::fibers::fiber_specific_ptr<int> fiber_id;
+static thread_local int thread_id;
+
+static std::mutex print_mutex;
+static std::size_t fiber_count{ 0 };
+static std::mutex count_mutex{};
+typedef std::unique_lock<std::mutex> lock_type;
```

Comment:
Be explicit and use `std::unique_lock<std::mutex>`. This code is already quite dense. No need to make it more convoluted.